src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1799,13 +1814,18 @@ def _retrieve_segment(
             segments = []
             if single_timestamp_ending:
                 slices.append(len(seek_sequence))
+            else:
+                # we want to include the last timestamp token in the last segment to know it was no single ending
+                slices[-1] += 1
 
             last_slice = 0
             # Add each segment to list of all segments
-            for current_slice in slices:
+            for i, current_slice in enumerate(slices):
+                is_last_slice = i == len(slices) - 1
                 sliced_tokens = seek_sequence[last_slice:current_slice]
                 start_timestamp_pos = sliced_tokens[0].item() - timestamp_begin
-                end_timestamp_pos = sliced_tokens[-1].item() - timestamp_begin
+                idx_sliced_tokens = -1 if not is_last_slice or single_timestamp_ending else -2
+                end_timestamp_pos = sliced_tokens[idx_sliced_tokens].item() - timestamp_begin
```

Comment:
not sure if relevant but each call to `.item()` will do a device synch, maybe calling `.item()` once after loop will be better?

Reply:
totally relevant, thanks !! This was unnecessary and is fixed in [5fba3e0](https://github.com/huggingface/transformers/pull/34537/commits/5fba3e0f0d9b422ce7b1aae9a527211749b89fce)